crates/core/src/nats/client.rs

```diff
@@ -0,0 +1,262 @@
+use std::{sync::Arc, time::Duration};
+
+use async_nats::connection;
+use bytes::Bytes;
+use tracing::info;
+
+use super::{
+    types::{
+        JetStreamConfig,
+        JetStreamContext,
+        NatsConsumer,
+        NatsStream,
+        PullConsumerConfig,
+    },
+    NatsError,
+    Subject,
+};
+use crate::types::BoxedResult;
+
+#[derive(Debug, Clone)]
+pub struct NatsClient {
+    pub url: String,
+    pub conn_id: String,
+    pub conn: Arc<async_nats::Client>,
+    pub jetstream: Arc<JetStreamContext>,
+}
+
+impl NatsClient {
+    pub async fn connect(
+        url: &str,
+        conn_id: &str,
+        nkey: Option<String>,
+    ) -> Result<Self, NatsError> {
+        let conn_id = conn_id.to_string();
+        let conn = Arc::new(Self::create_conn(url, nkey).await?);
+        let context =
+            Arc::new(async_nats::jetstream::new(conn.as_ref().clone()));
+
+        info!("Connected to NATS server at {}", url);
+        Ok(Self {
+            url: url.to_string(),
+            conn_id,
+            conn,
+            jetstream: context,
+        })
+    }
+
+    pub async fn validate_payload(
+        &self,
+        payload: Bytes,
+        subject_name: &str,
+    ) -> Result<(), NatsError> {
+        let payload_size = payload.len();
+        let max_payload_size = self.conn.server_info().max_payload;
+        if payload_size > max_payload_size {
+            return Err(NatsError::PayloadTooLarge {
+                subject: subject_name.to_string(),
+                payload_size,
+                max_payload_size,
+            });
+        }
+
+        Ok(())
+    }
```

Comment:
We should avoid defensively validating things that are outside our system. The NATS cluster is a separate system and should own the logic of validating the payload size since anyone can try that out.

In short, this doesn't defend anything but adds to the runtime cost. We should instead handle this error on publish and report it appropriately.

Reply:
Ok, I removed it from the `publish` inside the `NatsClient` and just left the method here so you can use it later in the Publisher within `fuel-core-nats`. Is this what you meant to do?